### ISSUE TITLE
UiTabs @beforeTabChange/@tryTabChange

### ISSUE THREAD

**Issue by Iyashi** [JosephusPaye/Keen-UI#407]
Hey,

I'm missing an event like `before-tab-change` for `UiTabs` since the `tab-change` event fires _after_ tab has successfully changed.

Example use case:
```
<template>
  <ui-tabs @tab-change="afterTabChange" @before-tab-change="beforeTabChange">
    <ui-tab id="t1">
      <form>
        <input :name="el.name" v-for="el in formElements" @change="editInput(el, $event.target.value)" :value="el.value"/>
      </form>
    </ui-tab>
    <ui-tab id="t2"></ui-tab>
  </ui-tabs>
</template>

<script>
export default {
  data () { return {
    formElements: [
      { name: 'id', placeholder:'#ID', value: null },
      /* ... */
    ]
  } },
  computed: {
    changed () { return this.formElements.filter($ => $.value !== null).length !== 0 },
  },
  methods: {
    editInput (el, value) {
      el.value = value && value.length ? value : null
    },
    beforeTabChange (id) {
      if (this.changed && window.confirm('Unsaved changes w/be destroyed. Continue?')) {
        this.tabChanged(id)
      }
      // additionally there could be a logic with `return true/false` (true: change, false: prevent)
      // so that you dont have to this.tabChange(id) yourself here
    },
    afterTabChange (id) {
      console.log('Changing tab finished')
    }
  }
}
</script>
```

Wishes

**Comment by JosephusPaye** [comment]
Makes sense, I'll add this.

**Comment by JosephusPaye** [comment]
Hi @Iyashi,

I've added a new prop `confirmTabChange` (https://github.com/JosephusPaye/Keen-UI/commit/8244007ac42cc4c938959137767e1ea32ff1d07e) which is a function that will be called for confirmation when the user attempts to change tabs.

Example:

```vue
<ui-tabs fullwidth :confirm-tab-change="confirmTabChange">
    <ui-tab title="Tab 1">Tab 1 content</ui-tab>
    <ui-tab title="Tab 2">Tab 2 content</ui-tab>
    <ui-tab title="Tab 3">Tab 3 content</ui-tab>
</ui-tabs>
```

```js
// in methods
confirmTabChange(currentTab, newTab) {
    return window.confirm('Are you sure you want to switch from ' + currentTab.title + ' to ' + newTab.title + '?');
}
```

**Comment by JosephusPaye** [comment]
Released in [v1.1.0](https://github.com/JosephusPaye/Keen-UI/releases/tag/v1.1.0).
